test-util/mock-http.ts

```diff
@@ -0,0 +1,125 @@
+import fs from 'fs';
+import path from 'path';
+import { DestinationAuthToken, HttpDestination, ServiceCredentials } from '@sap-cloud-sdk/connectivity';
+import nock from 'nock';
+import {
+  BaseLlmParameters,
+  CustomRequestConfig
+} from '@sap-ai-sdk/core';
+import { EndpointOptions } from '@sap-ai-sdk/core/src/http-client.js';
+import { dummyToken } from './mock-jwt.js';
+
+export const aiCoreDestination = {
+  url: 'https://api.ai.ml.hana.ondemand.com'
+};
+
+export const aiCoreServiceBinding = {
+  label: 'aicore',
+  credentials: {
+    clientid: 'clientid',
+    clientsecret: 'clientsecret',
+    url: 'https://example.authentication.eu12.hana.ondemand.com',
+    identityzone: 'examplezone',
+    identityzoneid: 'examplezoneid',
+    appname: 'appname',
+    serviceurls: {
+      AI_API_URL: aiCoreDestination.url
+    }
+  }
+};
+
+const mockEndpoint: EndpointOptions = {
+  url: 'mock-endpoint',
+  apiVersion: 'mock-api-version'
+};
+
+export function mockAiCoreEnvVariable(): void {
+  process.env['aicore'] = JSON.stringify(aiCoreServiceBinding.credentials);
+}
+
+export function createDestinationTokens(
+  token: string = dummyToken,
+  expiresIn?: string
+): { authTokens: DestinationAuthToken[] } {
+  return {
+    authTokens: [
+      {
+        value: token,
+        type: 'bearer',
+        expiresIn,
+        http_header: { key: 'Authorization', value: `Bearer ${token}` },
```

Comment:
[pp] As headers are case insensitive I would prefer to keep them lower case everywhere in our code. Totally just personal preference ;) 
```suggestion
        http_header: { key: 'authorization', value: `Bearer ${token}` },
```